pkg/controller/monitor/monitor_controller.go

```diff
@@ -191,6 +199,45 @@ func (r *ReconcileMonitor) Reconcile(ctx context.Context, request reconcile.Requ
 		}
 	}
 
+	var tlsSecret *corev1.Secret
+	if install.CertificateManagement == nil {
+		// Check that if the apiserver cert pair secret exists that it is valid (has key and cert fields)
+		// If it does not exist then this function still returns true
+		tlsSecret, err = utils.ValidateCertPair(r.client,
+			common.OperatorNamespace(),
+			monitor.PrometheusTLSSecretName,
+			corev1.TLSPrivateKeyKey,
+			corev1.TLSCertKey,
+		)
+		if err != nil {
+			log.Error(err, "Invalid TLS Cert")
+			r.status.SetDegraded("Error validating TLS certificate", err.Error())
+			return reconcile.Result{}, err
+		}
+		r.status.RemoveCertificateSigningRequests(common.TigeraPrometheusNamespace)
```

Comment:
Is for the case that somebody switches from certificate management to no certificate manager?

Reply:
Yes, that's right. It is then no longer necessary to watch for pending csr's in this namespace.